src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -387,7 +344,7 @@ private bool DoesPathPassNameFilter(string eventPath)
             Interop.EventStream.FSEventStreamEventFlags[] eventFlags,
             FSEventStreamEventId[] eventIds)
         {
-            Debug.Assert((currentIndex + 1 < eventPaths.Length) && (currentIndex + 1 < eventFlags.Length) & (currentIndex + 1 < eventIds.Length));
+            //Debug.Assert((currentIndex + 1 < eventPaths.Length) && (currentIndex + 1 < eventFlags.Length) && (currentIndex + 1 < eventIds.Length));
```

Comment:
The previous assumptions about this assert were incorrect in that we can have a rename without a pair when an item is moved out / in from / to the watcher directory. The current index may be the last item in the array of events, meaning the only thing we could assert is that the index is within the bounds of the array, but we would have other (more fundamental issues) if that ever occurred. I should remove the line though...forgot to do that